trie/trie.go

```diff
@@ -439,11 +447,13 @@ func (t *Trie) insert(n node, prefix, key []byte, value node, epoch types.StateE
 		// If the whole key matches, keep this short node as is
 		// and only update the value.
 		if matchlen == len(n.Key) {
-			n.setEpoch(t.currentEpoch)
 			dirty, nn, err := t.insert(n.Val, append(prefix, key[:matchlen]...), key[matchlen:], value, n.epoch)
 			if !dirty || err != nil {
 				return false, n, err
 			}
+			if t.withShadowNodes {
```

Comment:
may need not update epoch, it reinit a shortNode with `t.currentEpoch`

Reply:
resolve